src/resolvers/Mutation/auth.js

```diff
@@ -0,0 +1,48 @@
+const { google } = require('../../services/google');
+const jwt = require('jsonwebtoken');
+
+// Helpers.
+
+async function getPrismaUser({ db }, googleId) {
+  return await db.query.user({ where: { googleId } })
+}
+
+async function createPrismaUser({ db }, googleUser) {
+
+  if (!googleUser.hd || googleUser.hd !== 'framgia.com') {
+    throw new Error('Not allow access!');
+  }
+
+  return await db.mutation.createUser({
+    data: {
+      name: googleUser.name,
+      googleId: googleUser.id,
+      email: googleUser.email,
+      avatar: googleUser.picture
+    }
+  });
+}
+
+// Resolvers.
+
+const auth = {
+  authenticate: async (parent, { googleCode }, ctx, info) => {
```

Comment:
> What do argumentsparent and info do?

Wow, the parent is pointing to the root resolver, because any field in the schema has their resolver.
You can check this doc for more detail: https://www.prisma.io/blog/graphql-server-basics-the-schema-ac5e2950214e/#9d03

> Seems like not used in the arrow function, so any special reason for passing these two arguments?

`info` is the object data that after you authenticate, it will return for you.
In this case, `info` represent for

`
data {
  user:  {
     id:  ....
     name: .....
  }
token: aslkhfaslkfashflashfalshf
}
`

Reply:
OK, Thanks!